Modules/_zoneinfo.c

```diff
@@ -821,7 +828,7 @@ zoneinfo_ZoneInfo__unpickle_impl(PyTypeObject *type, PyTypeObject *cls,
             return NULL;
         }
 
-        PyObject *rv = zoneinfo_new(type, val_args, NULL);
+        PyObject *rv = zoneinfo_ZoneInfo(type, val_args, NULL);
```

Comment:
You can actually use `zoneinfo_ZoneInfo_impl(type, key)` (with critical section). `val_args` will no longer be needed.